### PR TITLE
Minor fixes in quickstart guide, fix rendering of $ in docs

### PR DESCRIPTION
**Pull request by nilsvu**
### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

**Comment by nilsvu** [comment]
@nilsdeppe this should fix #4036

File: cmake/FindCatch.cmake [hunk 1]
```diff
@@ -29,7 +29,7 @@ endfunction()
 
 find_path(
   CATCH_INCLUDE_DIR
-  PATH_SUFFIXES single_include include catch catch2
+  PATH_SUFFIXES single_include include catch catch2 include/catch2
```
Comment (wthrowe) [inline cmake/FindCatch.cmake:32]: Re commit message:
> Spack installs the headers in include/catch2.

If I'm interpreting this right that sounds like a Spack bug.  Is there an upstream issue or anything?

Comment (nilsvu) [reply cmake/FindCatch.cmake:32]: Is it a bug? Isn't it just fine to install in `include/catch2` so headers are included as `catch2/*.hpp`?

Comment (wthrowe) [reply cmake/FindCatch.cmake:32]: OK, I guess it's CMake stupid, then.  (That's the completely standard path and how I've always had it installed, so I assumed it was installing to a nested `include` or something.)

**Review by wthrowe — APPROVED** [review]

**Pull request merged**